chromadb/auth/token_authn/__init__.py

```diff
@@ -119,9 +136,19 @@ def __init__(self, system: System) -> None:
         super().__init__(system)
         self._settings = system.settings
         if system.settings.chroma_auth_token_transport_header:
-            self._token_transport_header = TokenTransportHeader[
-                str(system.settings.chroma_auth_token_transport_header)
-            ]
+            allowed_token_headers = [header.value for header in TokenTransportHeader]
+            if (
+                system.settings.chroma_auth_token_transport_header
+                not in allowed_token_headers
+            ):
+                raise ValueError(
+                    f"Invalid token transport header: {system.settings.chroma_auth_token_transport_header}. "
+                    f"Must be one of {allowed_token_headers}"
+                )
+
```

Comment:
Since we duplicate this logic could we pull it into a function

Reply:
Done